### ISSUE TITLE
New cluster stuck in provisioning state

### ISSUE THREAD

**Issue by rmahajan1234** [rancher/rancher#29859]
Hi

I did search on this forum but could not find a working solution. 

Rancher version: 2.4.8   HA
Cluster type: Custom hosted on AWS, 
Docker version: 1.13.1
Kubectl version:
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:41:02Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}

I deployed Rancher with self certificates with a private CA option. The Rancher GUI came up and when I try to create a new cluster, it is stuck in provisioning state with an error ‘check etcd logs’. The etcd logs says:
tls: failed to verify client’s certificate: x509: certificate signed by unknown authority (possibly because of “crypto/rsa: verification error” while trying to verify candidate authority certificate “kube-ca”)", ServerName “”

Steps to reproduce:

cluster.yml:
cluster_name: Ram-Kube
ssh_key_path: /home/ec2-user/.ssh/id_rsa
nodes:

address: 10.0.9.205
internal_address: 10.0.9.205
user: ec2-user
role: [controlplane,worker,etcd]
address: 10.0.9.197
internal_address: 10.0.9.197
user: ec2-user
role: [worker]
address: 10.0.10.177
internal_address: 10.0.10.177
user: ec2-user
role: [worker]
services:
etcd:
snapshot: true
creation: 6h
retention: 24h

network:
plugin: weave

ingress:
provider: nginx
options:
use-forwarded-headers: ‘true’

rke up --config ./cluster.yml
kubectl create namespace cattle-system
kubectl -n cattle-system create secret tls tls-rancher-ingress --cert=tls.crt --key=tls.key
kubectl -n cattle-system create secret generic tls-ca --from-file=cacerts.pem=./cacerts.pem
kubectl -n cattle-system create secret generic tls-ca-additional --from-file=ca-additional.pem=./ca-additional.pem

helm install rancher rancher-stable/rancher --namespace cattle-system --set hostname=rancher-draco.asc-dev.io --set ingress.tls.source=secret --set privateCA=true --kubeconfig ./kube_config_cluster.yml --set additionalTrustedCAs=true
Do you see the incorrect configuration parameter in the steps above?

Thanks in advance.
Ram

**Comment by superseb** [comment]
The error usually indicates that the node(s) that you are adding to the cluster created in Rancher have old files from a previous try (another setup or cluster). Is the node you are trying to add newly created and not previously used? Are you using the custom cluster (from existing nodes) or are you using a node driver to create nodes from a cloud provider (EC2/DigitalOcean/etc)?

**Comment by superseb** [comment]
Please share the output of the following commands from the node you are trying to add to the created custom cluster:

* `docker ps -a`
* `docker logs etcd` (full logs)
* `ls -la /etc/kubernetes/ssl`
* `ls -ld /var/lib/etcd`

The custom cluster created in Rancher is totally separate from the cluster running Rancher, so not sure what `This command got rid of etcd's error` exactly means. Can you also confirm that the custom cluster is newly created, and hasn't been created earlier then when you were testing.

**Comment by rmahajan1234** [comment]
'rke cert rotate --rotate-ca' has thrown my installation into an unstable one. I will terminate all Ec2 instances, recreate them, and run the install steps described in my first post and then provide the clean logs you requested. Since I have done this process now about 8 times, shouldn't take that long.

**Comment by rmahajan1234** [comment]
docker ps -a output:

`CONTAINER ID        IMAGE                                                                                                      COMMAND                  CREATED              STATUS                      PORTS               NAMES
a00b8657d43f        rancher/coreos-etcd:v3.4.3-rancher1                                                                        "/usr/local/bin/et..."   About a minute ago   Up About a minute                               etcd
eb2926b850cb        rancher/rancher-agent:v2.4.8                                                                               "run.sh --server h..."   2 minutes ago        Up 2 minutes                                    thirsty_agnesi
c8381660d07d        rancher/rancher-agent@sha256:c8a111e6250a313f1dd5d34696ddbef9068f70ddf4b15ab4c9cefd0ea39b76c1              "run.sh"                 3 minutes ago        Up 3 minutes                                    k8s_cluster-register_cattle-cluster-agent-7967f8544d-22mdv_cattle-system_0ff1661b-e51e-46ba-9423-9f03a0e3e339_0
a1aef2320782        rancher/rancher-agent@sha256:c8a111e6250a313f1dd5d34696ddbef9068f70ddf4b15ab4c9cefd0ea39b76c1              "run.sh"                 3 minutes ago        Up 3 minutes                                    k8s_agent_cattle-node-agent-5nmvg_cattle-system_86f4548d-fe76-4939-a35b-ec3716612ad9_0
3668b52ebbd7        rancher/pause:3.1                                                                                          "/pause"                 3 minutes ago        Up 3 minutes                                    k8s_POD_cattle-node-agent-5nmvg_cattle-system_86f4548d-fe76-4939-a35b-ec3716612ad9_0
5adaa897e57a        rancher/pause:3.1                                                                                          "/pause"                 3 minutes ago        Up 3 minutes                                    k8s_POD_cattle-cluster-agent-7967f8544d-22mdv_cattle-system_0ff1661b-e51e-46ba-9423-9f03a0e3e339_0
e953ed61be38        rancher/rancher@sha256:5a16a6a0611e49d55ff9d9fbf278b5ca2602575de8f52286b18158ee1a8a5963                    "entrypoint.sh --h..."   5 minutes ago        Up 5 minutes                                    k8s_rancher_rancher-76d84f5d78-ggxxq_cattle-system_9ca3e63f-8dfc-408f-9713-a7dd73f7cb91_0
ed150bc9223d        rancher/pause:3.1                                                                                          "/pause"                 6 minutes ago        Up 6 minutes                                    k8s_POD_rancher-76d84f5d78-ggxxq_cattle-system_9ca3e63f-8dfc-408f-9713-a7dd73f7cb91_0
92459dc4cc54        rancher/coredns-coredns@sha256:e70c936deab8efed89db66f04847fec137dbb81d5b456e8068b6e71cb770f6c0            "/coredns -conf /e..."   8 minutes ago        Up 8 minutes                                    k8s_coredns_coredns-849545576b-nm7bg_kube-system_84496dab-e4b0-4838-b49d-d00dd5f5e8c2_0
76f9036ee1b9        rancher/nginx-ingress-controller@sha256:a476350cfec645f316b1bf56de0240db7f81e85e97dbcb0a6d0984308c252d72   "/usr/bin/dumb-ini..."   8 minutes ago        Up 8 minutes                                    k8s_nginx-ingress-controller_nginx-ingress-controller-fkzbj_ingress-nginx_399763f3-6d02-45bd-b373-ca055420d1ac_0
4c6f1712530f        rancher/pause:3.1                                                                                          "/pause"                 8 minutes ago        Up 8 minutes                                    k8s_POD_coredns-849545576b-nm7bg_kube-system_84496dab-e4b0-4838-b49d-d00dd5f5e8c2_0
d20e7bcaadd5        rancher/pause:3.1                                                                                          "/pause"                 8 minutes ago        Up 8 minutes                                    k8s_POD_nginx-ingress-controller-fkzbj_ingress-nginx_399763f3-6d02-45bd-b373-ca055420d1ac_0
9ffbafee59f3        3f74dabda6a0                                                                                               "kubectl apply -f ..."   8 minutes ago        Exited (0) 8 minutes ago                        k8s_rke-ingress-controller-pod_rke-ingress-controller-deploy-job-kn2pr_kube-system_215b93c0-730f-4d22-95a5-7445584e64ab_0
d56b46142e8e        rancher/pause:3.1                                                                                          "/pause"                 8 minutes ago        Exited (0) 8 minutes ago                        k8s_POD_rke-ingress-controller-deploy-job-kn2pr_kube-system_215b93c0-730f-4d22-95a5-7445584e64ab_0
104c6c3b1d14        1df47b3cc6ea                                                                                               "/opt/rke-tools/we..."   8 minutes ago        Up 8 minutes                                    k8s_weave-plugins_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
e0e0b90bf954        weaveworks/weave-npc@sha256:9055fee731a5e902cf6f793af498d66ccf7442a7fe2e9fdf2f0f2401df538690               "/usr/bin/launch.sh"     8 minutes ago        Up 8 minutes                                    k8s_weave-npc_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
8005adaaf751        3f74dabda6a0                                                                                               "kubectl apply -f ..."   8 minutes ago        Exited (0) 8 minutes ago                        k8s_rke-metrics-addon-pod_rke-metrics-addon-deploy-job-jjshd_kube-system_6e18e111-c904-4a4e-85a3-6e2c03f2a6e3_0
0f0e65e66f9c        rancher/pause:3.1                                                                                          "/pause"                 8 minutes ago        Exited (0) 8 minutes ago                        k8s_POD_rke-metrics-addon-deploy-job-jjshd_kube-system_6e18e111-c904-4a4e-85a3-6e2c03f2a6e3_0
0f36bdb3674e        weaveworks/weave-kube@sha256:0176bd9994ddfd7ebacc95ae4885c2178a964c206ccd77bcc6c2fb0be53fe638              "/home/weave/launc..."   8 minutes ago        Up 8 minutes                                    k8s_weave_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
08a65130c47d        3f74dabda6a0                                                                                               "kubectl apply -f ..."   8 minutes ago        Exited (0) 8 minutes ago                        k8s_rke-coredns-addon-pod_rke-coredns-addon-deploy-job-24g8w_kube-system_041c3808-058f-4db8-84df-1d0ea46fe54b_0
f22e9c6560ee        rancher/pause:3.1                                                                                          "/pause"                 8 minutes ago        Exited (0) 8 minutes ago                        k8s_POD_rke-coredns-addon-deploy-job-24g8w_kube-system_041c3808-058f-4db8-84df-1d0ea46fe54b_0
3b9fa7482ce5        rancher/pause:3.1                                                                                          "/pause"                 8 minutes ago        Up 8 minutes                                    k8s_POD_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
0e85dbebe904        3f74dabda6a0                                                                                               "kubectl apply -f ..."   8 minutes ago        Exited (0) 8 minutes ago                        k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-dt866_kube-system_86b426ba-d6fd-492d-bf6e-80567c5ce576_0
3c5f083c8413        rancher/pause:3.1                                                                                          "/pause"                 8 minutes ago        Exited (0) 8 minutes ago                        k8s_POD_rke-network-plugin-deploy-job-dt866_kube-system_86b426ba-d6fd-492d-bf6e-80567c5ce576_0
3e4ee34175fe        rancher/hyperkube:v1.18.9-rancher1                                                                         "/opt/rke-tools/en..."   8 minutes ago        Up 8 minutes                                    kube-proxy
a703ed729b9c        rancher/hyperkube:v1.18.9-rancher1                                                                         "/opt/rke-tools/en..."   9 minutes ago        Up 2 minutes                                    kubelet
0d8a85acec2c        rancher/hyperkube:v1.18.9-rancher1                                                                         "/opt/rke-tools/en..."   9 minutes ago        Up 9 minutes                                    kube-scheduler
cbf11fce0c9c        rancher/hyperkube:v1.18.9-rancher1                                                                         "/opt/rke-tools/en..."   9 minutes ago        Up 9 minutes                                    kube-controller-manager
8362cabd680e        rancher/hyperkube:v1.18.9-rancher1                                                                         "/opt/rke-tools/en..."   9 minutes ago        Up 9 minutes                                    kube-apiserver
4a6128d8a0b3        rancher/rke-tools:v0.1.65                                                                                  "/bin/bash"              9 minutes ago        Created                                         service-sidekick
0e4027050186        rancher/rke-tools:v0.1.65                                                                                  "/docker-entrypoin..."   10 minutes ago       Exited (0) 10 minutes ago                       cluster-state-deployer
`
====================
docker logs etcd:
`2020-10-30 20:39:51.069065 W | pkg/flags: unrecognized environment variable ETCD_UNSUPPORTED_ARCH=x86_64
[WARNING] Deprecated '--logger=capnslog' flag is set; use '--logger=zap' flag instead
2020-10-30 20:39:51.069161 I | etcdmain: etcd Version: 3.4.3
2020-10-30 20:39:51.069175 I | etcdmain: Git SHA: 3cf2f69b5
2020-10-30 20:39:51.069188 I | etcdmain: Go Version: go1.12.12
2020-10-30 20:39:51.069201 I | etcdmain: Go OS/Arch: linux/amd64
2020-10-30 20:39:51.069214 I | etcdmain: setting maximum number of CPUs to 2, total number of available CPUs is 2
2020-10-30 20:39:51.069305 N | etcdmain: the server is already initialized as member before, starting as etcd member...
[WARNING] Deprecated '--logger=capnslog' flag is set; use '--logger=zap' flag instead
2020-10-30 20:39:51.069348 I | embed: peerTLS: cert = /etc/kubernetes/ssl/kube-etcd-10-0-11-223.pem, key = /etc/kubernetes/ssl/kube-etcd-10-0-11-223-key.pem, trusted-ca = /etc/kubernetes/ssl/kube-ca.pem, client-cert-auth = true, crl-file =
2020-10-30 20:39:51.070226 I | embed: name = etcd-ip-10-0-11-223
2020-10-30 20:39:51.070244 I | embed: data dir = /var/lib/rancher/etcd/
2020-10-30 20:39:51.070258 I | embed: member dir = /var/lib/rancher/etcd/member
2020-10-30 20:39:51.070271 I | embed: heartbeat = 500ms
2020-10-30 20:39:51.070283 I | embed: election = 5000ms
2020-10-30 20:39:51.070296 I | embed: snapshot count = 100000
2020-10-30 20:39:51.070315 I | embed: advertise client URLs = https://10.0.11.223:2379,https://10.0.11.223:4001
2020-10-30 20:39:51.070329 I | embed: initial advertise peer URLs = https://10.0.11.223:2380
2020-10-30 20:39:51.070343 I | embed: initial cluster =
2020-10-30 20:39:51.124318 I | etcdserver: restarting member e6f27406986f4bd2 in cluster f6ed3aa5735e535b at commit index 4603
raft2020/10/30 20:39:51 INFO: e6f27406986f4bd2 switched to configuration voters=()
raft2020/10/30 20:39:51 INFO: e6f27406986f4bd2 became follower at term 2
raft2020/10/30 20:39:51 INFO: newRaft e6f27406986f4bd2 [peers: [], term: 2, commit: 4603, applied: 0, lastindex: 4603, lastterm: 2]
2020-10-30 20:39:51.140547 W | auth: simple token is not cryptographically signed
2020-10-30 20:39:51.142506 I | etcdserver: starting server... [version: 3.4.3, cluster version: to_be_decided]
raft2020/10/30 20:39:51 INFO: e6f27406986f4bd2 switched to configuration voters=(16641491144763460562)
2020-10-30 20:39:51.144512 I | embed: ClientTLS: cert = /etc/kubernetes/ssl/kube-etcd-10-0-11-223.pem, key = /etc/kubernetes/ssl/kube-etcd-10-0-11-223-key.pem, trusted-ca = /etc/kubernetes/ssl/kube-ca.pem, client-cert-auth = true, crl-file =
2020-10-30 20:39:51.144557 I | etcdserver/membership: added member e6f27406986f4bd2 [https://10.0.11.223:2380] to cluster f6ed3aa5735e535b
2020-10-30 20:39:51.144684 N | etcdserver/membership: set the initial cluster version to 3.4
2020-10-30 20:39:51.144748 I | etcdserver/api: enabled capabilities for version 3.4
2020-10-30 20:39:51.144887 I | embed: listening for peers on [::]:2380
raft2020/10/30 20:39:58 INFO: e6f27406986f4bd2 is starting a new election at term 2
raft2020/10/30 20:39:58 INFO: e6f27406986f4bd2 became candidate at term 3
raft2020/10/30 20:39:58 INFO: e6f27406986f4bd2 received MsgVoteResp from e6f27406986f4bd2 at term 3
raft2020/10/30 20:39:58 INFO: e6f27406986f4bd2 became leader at term 3
raft2020/10/30 20:39:58 INFO: raft.node: e6f27406986f4bd2 elected leader e6f27406986f4bd2 at term 3
2020-10-30 20:39:58.627364 I | etcdserver: published {Name:etcd-ip-10-0-11-223 ClientURLs:[https://10.0.11.223:2379 https://10.0.11.223:4001]} to cluster f6ed3aa5735e535b
2020-10-30 20:39:58.627596 I | embed: ready to serve client requests
2020-10-30 20:39:58.628978 I | embed: serving client requests on [::]:2379
2020-10-30 20:39:58.839060 I | embed: rejected connection from "10.0.11.223:55148" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:40:03.868089 I | embed: rejected connection from "10.0.11.223:55490" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:40:08.887993 I | embed: rejected connection from "10.0.11.223:55576" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:40:21.021257 I | embed: rejected connection from "10.0.11.223:38455" (error "EOF", ServerName "")
2020-10-30 20:40:28.966778 I | embed: rejected connection from "10.0.11.223:55408" (error "read tcp 10.0.11.223:2379->10.0.11.223:55408: read: connection reset by peer", ServerName "")
2020-10-30 20:40:28.966863 I | embed: rejected connection from "10.0.11.223:55406" (error "read tcp 10.0.11.223:2379->10.0.11.223:55406: read: connection reset by peer", ServerName "")
2020-10-30 20:40:33.625030 I | embed: rejected connection from "10.0.11.223:55964" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:40:38.641513 I | embed: rejected connection from "10.0.11.223:56036" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:40:43.656338 I | embed: rejected connection from "10.0.11.223:56062" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:41:55.428194 I | embed: rejected connection from "10.0.11.223:39280" (error "EOF", ServerName "")
2020-10-30 20:42:07.067192 I | embed: rejected connection from "10.0.11.223:56452" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:42:12.084309 I | embed: rejected connection from "10.0.11.223:56538" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:42:17.099335 I | embed: rejected connection from "10.0.11.223:56620" (error "tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kube-ca\")", ServerName "")
2020-10-30 20:42:19.111571 I | mvcc: store.index: compact 3368
2020-10-30 20:42:19.162858 I | mvcc: finished scheduled compaction at 3368 (took 50.532099ms)
`
==================================
[root@ip-10-0-11-223 ec2-user]# ls -la /etc/kubernetes/ssl
total 108
drwxr-xr-x. 3 root root 4096 Oct 30 20:38 .
drwxr-xr-x. 3 root root   77 Oct 30 20:44 ..
drwx------. 2 root root   22 Oct 30 20:38 certs
-rw-------. 1 root root 1679 Oct 30 20:31 kube-apiserver-key.pem
-rw-------. 1 root root 1241 Oct 30 20:31 kube-apiserver.pem
-rw-------. 1 root root 1679 Oct 30 20:31 kube-apiserver-proxy-client-key.pem
-rw-------. 1 root root 1107 Oct 30 20:31 kube-apiserver-proxy-client.pem
-rw-------. 1 root root 1679 Oct 30 20:31 kube-apiserver-requestheader-ca-key.pem
-rw-------. 1 root root 1082 Oct 30 20:31 kube-apiserver-requestheader-ca.pem
-rw-------. 1 root root 1679 Oct 30 20:31 kube-ca-key.pem
-rw-------. 1 root root 1017 Oct 30 20:31 kube-ca.pem
-rw-------. 1 root root  517 Oct 30 20:31 kubecfg-kube-apiserver-proxy-client.yaml
-rw-------. 1 root root  533 Oct 30 20:31 kubecfg-kube-apiserver-requestheader-ca.yaml
-rw-------. 1 root root  501 Oct 30 20:31 kubecfg-kube-controller-manager.yaml
-rw-------. 1 root root  445 Oct 30 20:31 kubecfg-kube-node.yaml
-rw-------. 1 root root  449 Oct 30 20:31 kubecfg-kube-proxy.yaml
-rw-------. 1 root root  465 Oct 30 20:31 kubecfg-kube-scheduler.yaml
-rw-------. 1 root root 1679 Oct 30 20:31 kube-controller-manager-key.pem
-rw-------. 1 root root 1062 Oct 30 20:31 kube-controller-manager.pem
-rw-------. 1 root root 1675 Oct 30 20:31 kube-etcd-10-0-11-223-key.pem
-rw-------. 1 root root 1233 Oct 30 20:31 kube-etcd-10-0-11-223.pem
-rw-------. 1 root root 1679 Oct 30 20:31 kube-node-key.pem
-rw-------. 1 root root 1070 Oct 30 20:31 kube-node.pem
-rw-------. 1 root root 1679 Oct 30 20:31 kube-proxy-key.pem
-rw-------. 1 root root 1046 Oct 30 20:31 kube-proxy.pem
-rw-------. 1 root root 1679 Oct 30 20:31 kube-scheduler-key.pem
-rw-------. 1 root root 1050 Oct 30 20:31 kube-scheduler.pem
-rw-------. 1 root root 1679 Oct 30 20:31 kube-service-account-token-key.pem
-rw-------. 1 root root 1241 Oct 30 20:31 kube-service-account-token.pem
=================
[root@ip-10-0-11-223 ec2-user]# ls -ld /var/lib/etcd
drwx------. 3 root root 20 Oct 30 20:39 /var/lib/etcd
======================
The custom cluster is newly created. I installed everything from scratch including brand new EC2 instances.

**Comment by rmahajan1234** [comment]
[ec2-user@ip-10-0-11-223 ~]$ docker ps -a | awk {'print $10'}

etcd
19
k8s_cluster-register_cattle-cluster-agent-7967f8544d-22mdv_cattle-system_0ff1661b-e51e-46ba-9423-9f03a0e3e339_0
k8s_agent_cattle-node-agent-5nmvg_cattle-system_86f4548d-fe76-4939-a35b-ec3716612ad9_0
k8s_POD_cattle-node-agent-5nmvg_cattle-system_86f4548d-fe76-4939-a35b-ec3716612ad9_0
k8s_POD_cattle-cluster-agent-7967f8544d-22mdv_cattle-system_0ff1661b-e51e-46ba-9423-9f03a0e3e339_0
minutes
k8s_POD_rancher-76d84f5d78-ggxxq_cattle-system_9ca3e63f-8dfc-408f-9713-a7dd73f7cb91_0
25
k8s_nginx-ingress-controller_nginx-ingress-controller-fkzbj_ingress-nginx_399763f3-6d02-45bd-b373-ca055420d1ac_0
k8s_POD_coredns-849545576b-nm7bg_kube-system_84496dab-e4b0-4838-b49d-d00dd5f5e8c2_0
k8s_POD_nginx-ingress-controller-fkzbj_ingress-nginx_399763f3-6d02-45bd-b373-ca055420d1ac_0
Exited
minutes
k8s_weave-plugins_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
k8s_weave-npc_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
Exited
minutes
k8s_weave_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
Exited
minutes
k8s_POD_weave-net-lvn8t_kube-system_3859ad02-ebf7-4782-afba-9c8f35bb1530_0
Exited
minutes
kube-proxy
kubelet
kube-scheduler
kube-controller-manager
kube-apiserver

**Comment by superseb** [comment]
Part of the issue is this line: ```2020-10-30 20:39:51.069305 N | etcdmain: the server is already initialized as member before, starting as etcd member...
``` which indicates this is not a new node as it finds old data. I guess we need the output of `find /var/lib/etcd -ls`.

**Comment by rmahajan1234** [comment]
[root@ip-10-0-11-223 ec2-user]# find /var/lib/etcd -ls
13160834    0 drwx------   3 root     root           20 Oct 30 20:39 /var/lib/etcd
121660915    0 drwx------   4 root     root           29 Oct 30 20:39 /var/lib/etcd/member
125875024    0 drwx------   2 root     root          108 Oct 31 12:40 /var/lib/etcd/member/snap
125875025 12764 -rw-------   1 root     root     13070336 Oct 31 14:21 /var/lib/etcd/member/snap/db
126034741    8 -rw-r--r--   1 root     root         7652 Oct 31 04:28 /var/lib/etcd/member/snap/0000000000000003-00000000000186a1.snap
126034743    8 -rw-r--r--   1 root     root         7652 Oct 31 12:40 /var/lib/etcd/member/snap/0000000000000003-0000000000030d42.snap
1091028    0 drwx------   2 root     root          199 Oct 31 10:03 /var/lib/etcd/member/wal
1091044 62504 -rw-------   1 root     root     64003048 Oct 31 00:36 /var/lib/etcd/member/wal/0000000000000000-0000000000000000.wal
1091047 62504 -rw-------   1 root     root     64001624 Oct 31 05:19 /var/lib/etcd/member/wal/0000000000000001-000000000000ce4b.wal
1301023 62504 -rw-------   1 root     root     64000560 Oct 31 10:03 /var/lib/etcd/member/wal/0000000000000002-000000000001af78.wal
1091045 62500 -rw-------   1 root     root     64000000 Oct 31 10:03 /var/lib/etcd/member/wal/1.tmp
1885557 62500 -rw-------   1 root     root     64000000 Oct 31 14:21 /var/lib/etcd/member/wal/0000000000000003-00000000000290c1.wal

**Comment by superseb** [comment]
And can be include the log from the `rancher/rancher`, the timestamp from some etcd files are way before the time the cluster got provisioned? (`/etc/kubernetes/ssl`)

**Comment by rmahajan1234** [comment]
Timestamp on /etc/kubernetes/ssl is        Oct 30 20:31
Timestamp on the first line of etcd log is Oct 30 20:39:51
-----------
The rancher/rancher container log file is huge. I do not see any way of attaching a file to this comment. The last few lines say this:
{"log":"2020/10/31 15:04:18 [WARNING] [etcd] host [10.0.11.223] failed to check etcd health: failed to get /health for host [10.0.11.223]: Get https://10.0.11.223:2379/health: remote error: tls: bad certificate\n","stream":"stdout","time":"2020-10-31T15:04:18.266226022Z"}
{"log":"2020/10/31 15:04:18 [ERROR] cluster [c-xtg22] provisioning: [etcd] Failed to bring up Etcd Plane: etcd cluster is unhealthy: hosts [10.0.11.223] failed to report healthy. Check etcd container logs on each host for more information\n","stream":"stdout","time":"2020-10-31T15:04:18.266576597Z"}
{"log":"2020/10/31 15:04:18 [INFO] kontainerdriver rancherkubernetesengine stopped\n","stream":"stdout","time":"2020-10-31T15:04:18.283491148Z"}
{"log":"2020/10/31 15:04:18 [ERROR] ClusterController c-xtg22 [cluster-provisioner-controller] failed with : [etcd] Failed to bring up Etcd Plane: etcd cluster is unhealthy: hosts [10.0.11.223] failed to report healthy. Check etcd container logs on each host for more information\n","stream":"stdout","time":"2020-10-31T15:04:18.371748949Z"}

**Comment by rmahajan1234** [comment]
Get https://10.0.11.223:2379/health: remote error: tls: bad certificate\n"
is referenced in another issue here: https://github.com/rancher/rke/issues/1244

Do I need to do the following:
# Remove your `<file>.rkestate` file
# Remove kube_config_cluster.yml
# Log into all of your control plane nodes and run:
rm -f /etc/kubernetes/ssl/kube-service-account-token-key.pem
rm -f /etc/kubernetes/ssl/kube-service-account-token.pem
cp /etc/kubernetes/ssl/kube-apiserver-key.pem /etc/kubernetes/ssl/kube-service-account-token-key.pem
cp /etc/kubernetes/ssl/kube-apiserver.pem /etc/kubernetes/ssl/kube-service-account-token.pem

and then try to add a cluster on Rancher GUI?

**Comment by superseb** [comment]
Look at the timestamps of the files/directories in `/var/lib/etcd`. Also, the outputs seem incorrect as on some `docker ps` outputs I see `rancher_rancher` containers which is the container from the pods that run Rancher on Kubernetes. The error is solely coming from the node that you are trying to add to the custom cluster. All outputs requested are from that node, not the nodes in the RKE installation for Rancher HA. The only way this can be related to the RKE installation for Rancher HA is if the cluster state stored in that cluster is wrong but that shouldnt happen if you create a new custom cluster every time.

So every action given around cleaning the node or recycling it, was regarding `ip-10-0-11-223`. Basically anything listed on https://rancher.com/docs/rancher/v2.x/en/cluster-admin/cleaning-cluster-nodes/ should be removed before creating a new cluster in Rancher and adding that node to the cluster. The etcd logging can be explained if cluster provisioning was kicked off a second time from Rancher, but for that we need the logs. Also, the timestamps are still confusing but that depends on what time the cluster was first provisioned from Rancher (which can also be retrieved from the logs)

**Comment by rmahajan1234** [comment]
Additional information:
- I have a bootstrap node from where I run rke and helm commands. This bootstrap node (10.0.13.235) is not part of the cluster. 
- In the cluster created by rke, there are one Master and 2 Worker nodes. Once the cluster is ready using rke command, I run helm commands to install Rancher on top of this Kubernetes cluster.
- Once I have access to the Rancher GUI, I log in as Admin, and click on 'Add cluster' button, and choose 'Existing Nodes' option and keep everything else as a default.
- The 10.0.11.223 is a Master node and the etcd log output is retrieved from the Master node. 
- Confusion: why do I need to clean the Master node? When I create a new cluster using Rancher GUI, does it need to be on a "brand new" node?

**Comment by rmahajan1234** [comment]
I think I finally got it. I need brand new EC2 instances to create a new cluster on (the wording 'Existing Nodes' is confusing though). 
The new cluster got created successfully and I was so happy, which lasted only a minute to see that Rancher GUI disappeared.
From the bootstrap node: kubectl get pods --all-namespaces returns
==> The connection to the server 10.0.11.223:6443 was refused - did you specify the right host or port?

**Comment by superseb** [comment]
Bringing up a cluster with a node does not influence the RKE cluster running Rancher HA. The steps provided on https://rancher.com/docs/rancher/v2.x/en/troubleshooting/ should be enough to check the status of the RKE cluster running Rancher HA.

Based on your outputs, `10.0.11.223` is not part of the Rancher HA cluster based on the `cluster.yml` from above, I also don't understand how this is being tested with kubectl.

This is how its supposed to be (in case of RKE):

- 3 Linux nodes with Docker installed, added to `rancher-cluster.yml`, provisioned using `rke up`
- DNS name pointing to a load balancer which load balances across those 3 nodes
- Install Rancher using Helm on this 3 node RKE cluster with hostname set to the DNS name pointing to the load balancer
- Login to Rancher using the configured hostname
- In the UI, create a new custom cluster, in the final screen, check all roles, and you get a command (`docker run`) to add nodes to your cluster
- Provision a new Linux node with Docker and run this command to add the node to the cluster
- Wait for Rancher to provision the cluster

RKE CLI and the created kubeconfig is how to interact (retrieve info/upgrade etc) with the RKE cluster running Rancher HA, the embedded kubectl shell or kubeconfig created with Rancher is how to interact with the custom cluster.

**Comment by rmahajan1234** [comment]
Please ignore the private ip addresses. Every single time I terminate the nodes and bring up the new nodes, just to have a clean slate.
1. I have 3 Linux nodes with a docker installed. In addition, I have a bootstrap node from where I run 'rke up' command. Do these 3 nodes need to have roles of [controlplane,worker,etcd] ?
2. DNS is all good.
3. The helm command to install Rancher is issued from the bootstrap node.
4. Rest of the commands are same.
5. The bootstrap node has a cluster.yml. It also gets kube_config_cluster.yml in the process. I run kubectl command from this bootstrap node:
kubectl get pods --all-namespaces --kubeconfig ./kube_config_cluster.yml

Question: Is it correct to have a separate bootstrap node to issue 'rke up' and helm commands?

**Comment by superseb** [comment]
For Rancher HA, yes it is advised to have 3 nodes with all roles. Its not technically required to have a bootstrap node, usually people use the machine they already work on (workstation or maybe a jumphost/admin server), but in this case it helps to keep everything separated and clear for the diagnosing the issue.


From the error `The connection to the server 10.0.11.223:6443 was refused`, log in to `10.0.11.223` and check node status and Docker status. The container running on here should be `kube-apiserver`, if it is running, check the logging. This is all covered in https://rancher.com/docs/rancher/v2.x/en/troubleshooting/

**Comment by superseb** [comment]
Let me know if this still needs investigating
